docs/content/python/getting-started/application/enforce.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Add authorization enforcement
+description: |
+    Oso policies define rules for who can perform actions on resources
+    in your application. The Oso library queries the policy to obtain an
+    authorization decision — a yes or no answer. You enforce that
+    decision by rejecting or accepting requests that users make.
+weight: 2
+---
+
+# Add authorization enforcement
+
+## The allow rule
+
+At the end of [Model your authorization policy](model) we defined an `allow`
+rule. The `allow` rule is used for resource-level enforcement. It
+accepts three arguments: an `actor` (who is making the request), an
+`action` (what the actor wants to do), and a `resource` (the object that the
+actor wants to perform the action on).
+
+Here's the `allow` rule you wrote:
+
+```polar
+allow(actor, action, resource) if
+	has_permission(actor, action, resource);
+```
+
+This rule succeeds if the `actor` `has_permission` to perform `action` on
+`resource`. `actor`, `action`, and `resource` are all variables. They
+are provided when you query the policy.
+
+## Querying the policy
+
+The Oso library queries the policy with inputs from your application. To
+construct a query, you give a rule name to query with a given list of
+parameters. The query returns a result for every rule that succeeds for
+the parameters specified. If there are multiple rules that succeed,
+multiple results will be returned. If no rules succeed, no results are
+returned from the query.
+
+TODO image of query
+
+To enforce authorization, we query for the `allow` rule with a specific `actor`,
+`action`, and `resource`.
+
+## `authorize`
+
+The `authorize` method queries the `allow` rule. If the query doesn't have any
+results (no rules succeed), it throws an `AuthorizationError`. You should handle
+this exception and [return an error response to the user]({{< ref "/guides/enforcement/resource#authorization-failure" >}}).
+
+The `authorize` method should be called any time you want to check if a uier can
```

Comment:
*user

Reply:
uier definitely gonna need to fix this one